platforms/ide/ide/src/main/java/org/gradle/plugins/ide/internal/tooling/GradleBuildBuilder.java

```diff
@@ -47,81 +48,98 @@ public boolean canBuild(String modelName) {
 
     @Override
     public DefaultGradleBuild create(BuildState target) {
-        target.ensureProjectsLoaded();
-        return convert(target, new LinkedHashMap<>());
+        return new GradleBuildCreator(target).create();
     }
 
-    private DefaultGradleBuild convert(BuildState targetBuild, Map<BuildState, DefaultGradleBuild> all) {
-        DefaultGradleBuild model = all.get(targetBuild);
-        if (model != null) {
-            return model;
+    @NullMarked
+    protected class GradleBuildCreator {
+        protected final BuildState target;
+        protected final Map<BuildState, DefaultGradleBuild> all = new LinkedHashMap<>();
```

Comment:
❓  Do we need this now as a field or we can revert it to previous logic where we didn't have a field?